core/src/main/java/org/infinispan/CacheStreamable.java

```diff
@@ -0,0 +1,120 @@
+package org.infinispan;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.BaseStream;
+
+/**
+ * Interface that signifies which streams are able to be used with a {@link Cache}.  This method
```

Comment:
This looks a bit hard to read, I'm not sure how a user could use a stream with a Cache.
s/This method/This interface/